### PR TITLE
Fixed renaming of auxiliary coordinates in `multi_model_statistics` if coordinates are equal

### LINKED ISSUE #1501: `multi_model_statistics` renames auxiliary coordinates
**Describe the bug**
The `multi_model_statistics` renames auxiliary coordinates. I'm not sure how often this happens, but it causes `recipe_kcs.yml` to fail (see ESMValGroup/ESMValTool#2223). In this specific case, the name `year` is removed from the coordinate:

Original cubes:
```
air_temperature / (K)               (time: 11)
    Dimension coordinates:
        time                             x
    Auxiliary coordinates:
        year                             x
...
```

Multi-model cubes:
```
air_temperature / (K)               (time: 11)
    Dimension coordinates:
        time                             x
    Auxiliary coordinates:
        unknown                          x
...
```

**Please attach**
https://esmvaltool.dkrz.de/shared/esmvaltool/v2.5.0-test-rc2/recipe_kcs_20220213_153930/run/main_log_debug.txt

### PR DESCRIPTION
**Pull request by schlunma**
## Description

This PR makes sure that coordinates that are equal in all input cubes are not renamed in `multi_model_statistics`. In addition, I cleaned the code a little bit (no further change in functionality).

Closes #1501

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by schlunma** [comment]
> I am a bit concerned that by removing the cell measures completely we break any other post-multimodel analysis - I can't remember, do we save individual cubes before the MM step or at the end of the MM step?

This is also done in the original code, but the cubes are copied before that:

https://github.com/ESMValGroup/ESMValCore/blob/d11b0461dfea043b8838aaf06c38d56e780483c3/esmvalcore/preprocessor/_multimodel.py#L344

**Comment by sloosvel** [comment]
>  I can't remember, do we save individual cubes before the MM step or at the end of the MM step?

MM was moved to the end of the chain

File: esmvalcore/preprocessor/_multimodel.py [hunk 1]
```diff
@@ -223,35 +223,71 @@ def _align(cubes, span):
...
-    """
-    equalise_attributes(cubes)  # in-place
-
-    for i, cube in enumerate(cubes):
-        concat_dim = iris.coords.AuxCoord(i, var_name=CONCAT_DIM)
+def _equalise_cell_methods(cubes):
+    """Equalise coordinates in cubes (in-place)."""
+    # Simply remove all cell methods
+    for cube in cubes:
+        cube.cell_methods = None
```
Comment (sloosvel) [inline esmvalcore/preprocessor/_multimodel.py:230]: What about ancillary variables? remove_fx_variables handles both.

Comment (schlunma) [reply esmvalcore/preprocessor/_multimodel.py:230]: I don't fully get this question. `remove_fx_variables` is called on all cubes in the old and new version of this. Sorry, the diff is a little bit crowded, but the only real change affects `equalise_coordinates`.

**Review by sloosvel — APPROVED** [review]

**Comment by schlunma** [comment]
Thanks :+1:

**Comment by valeriupredoi** [comment]
cheers guys, am in the middle of a hurricane-sized storm, so a bit harder to work :grin:

**Pull request merged**